src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,504 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import re
+from collections import defaultdict
+from functools import cached_property
+from typing import Optional, Union, List, TYPE_CHECKING
+
+from snowflake.snowpark._internal.data_source import DataSourcePartitioner
+from snowflake.snowpark._internal.data_source.drivers import BaseDriver
+from snowflake.snowpark._internal.utils import (
+    generate_random_alphanumeric,
+    random_name_for_temp_object,
+    TempObjectType,
+)
+from snowflake.snowpark.types import (
+    StructType,
+    DecimalType,
+    FloatType,
+    StringType,
+    DateType,
+    BooleanType,
+    ArrayType,
+    StructField,
+    VariantType,
+    TimestampType,
+    TimestampTimeZone,
+    IntegerType,
+    DoubleType,
+)
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session
+    from snowflake.snowpark.dataframe import DataFrame
+
+logger = logging.getLogger(__name__)
+
+# reference for java type to snowflake type:
+# https://docs.snowflake.com/en/developer-guide/udf-stored-procedure-data-type-mapping
+
+JAVA_TYPE_TO_SNOWFLAKE_TYPE = {
+    "java.math.BigDecimal": DecimalType,
+    "java.math.BigInteger": DecimalType,
+    "java.lang.Float": FloatType,
+    "java.lang.Double": FloatType,
+    "java.lang.Boolean": BooleanType,
+    "java.lang.String": StringType,
+    "java.sql.Timestamp": TimestampType,
+    "java.sql.Date": DateType,
+    "java.sql.Array": ArrayType,
+    "java.lang.String[]": ArrayType,
+    "java.lang.Long": DecimalType,
+    "java.lang.Integer": DecimalType,
+    "java.lang.Short": DecimalType,
+}
+JDBC_TYPE_TO_SNOWFLAKE_TYPE = {
```

Comment:
can we have a complete list here, looks like there are still a lot of primitives we miss here: https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html, I think there are some primitive types like Date -> DateType, Time -> TimeType

or do you use another newer reference doc?



ideally we should have similar structure like the following:
 https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/data_source/drivers/psycopg2_driver.py#L41-L116